### PR TITLE
Remove upper bounds for numpy.

### LINKED ISSUE #1837: Increase upper numpy version bound?
In https://github.com/databricks/koalas/pull/1734, it looks like the numpy upper bound was lowered to `<1.19.0`. Is there something in koalas that is incompatible with 1.19.x? If not, can it be increased to `<1.20.0`?

cc @ueshin

### PR DESCRIPTION
**Pull request by itholic**
This should resolve #1837

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -1484,6 +1484,16 @@ def delete(self, loc):
                     "index {} is out of bounds for axis 0 with size {}".format(loc, length)
                 )
             loc = [loc]
+        # pandas with numpy>=1.19.0 raises Exception if the item in the list exceeds the length.
+        # Because the behavior of `np.delete` has been changed from numpy 1.19.0
+        # which pandas uses directly.
+        elif LooseVersion(np.__version__) >= LooseVersion("1.19.0"):
```
Comment (ueshin) [inline databricks/koalas/indexes.py:1490]: I'm wondering whether we should switch the behavior based on the underlying numpy?
I guess we can just raise the `IndexError` when out of bounds?

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -1484,6 +1484,16 @@ def delete(self, loc):
...
+            for index in loc:
+                if abs(index) >= length:
```
Comment (ueshin) [inline databricks/koalas/indexes.py:1492]: Btw, this is not right?

```py
>>> pidx = pd.Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10])
>>> len(pidx)
12
>>> pidx.delete(-12)
Int64Index([10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
>>> pidx.delete([-12])
Int64Index([10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
>>> pidx.delete(-13)
Traceback (most recent call last):
...
IndexError: index -13 is out of bounds for axis 0 with size 12
>>> pidx.delete([-13])
Traceback (most recent call last):
...
IndexError: index -13 is out of bounds for axis 0 with size 12
```

whereas:

```py
>>> kidx = ks.from_pandas(pidx)
>>> kidx.delete(-12)
Traceback (most recent call last):
...
IndexError: index -12 is out of bounds for axis 0 with size 12
>>> kidx.delete([-12])
Traceback (most recent call last):
...
IndexError: index -12 is out of bounds for axis 0 with size 12
```

File: databricks/koalas/indexes.py (hunk 1 above)
Comment (HyukjinKwon) [reply databricks/koalas/indexes.py:1490]: +1

File: databricks/koalas/indexes.py (hunk 2 above)
Comment (HyukjinKwon) [reply databricks/koalas/indexes.py:1492]: Hey @itholic, let's make sure we check the corner cases before you make a PR.

Comment (itholic) [reply databricks/koalas/indexes.py:1492]: Thanks, @ueshin @HyukjinKwon .
I'll make a follow-up soon and will more careful before open up the PR.

**Pull request merged**